cmd/agent/command/command.go

```diff
@@ -86,7 +90,7 @@ monitoring and performance data.`,
 	// github.com/fatih/color sets its global color.NoColor to a default value based on
 	// whether the process is running in a tty.  So, we only want to override that when
 	// the value is true.
-	agentCmd.PersistentFlags().BoolVarP(&globalParams.NoColor, "no-color", "n", false, "disable color output")
+	agentCmd.PersistentFlags().BoolVarP(&globalParams.NoColor, "no-color", "n", globalParams.NoColor, "disable color output")
```

Comment:
This is setting the default value dynamically depending on whether the output is a tty, I think this is very error prone...
```
$ agent -h
[...]
  -n, --no-color                   disable color output

$ agent -h | less
[...]
  -n, --no-color                   disable color output (default true)
```

Why not just override `color.NoColor` with `true` just like we used to do ?

Reply:
To provide some context on why the color output broke in the first place. Prior to this PR https://github.com/DataDog/datadog-agent/pull/25595 the `configcheck` command worked by asking the server for the data and then it would iterate over the data and print to the terminal. We found that there could be potential issue with the scrubber. The scrubber in the client has no recollection of which keys used the secret resolver backend command, so we could display secret into the terminal. The PR above move the logic of rendering the result of `configcheck` into the server. It fixed the issue with potential secrets being displayed in the terminal but created the problem we are today in which the client and the server run in separate process. The client runs in a TTY and the `color.NoColor` is set to to true by the library https://github.com/fatih/color/blob/main/color.go#L22-L23 but we haven't specified the flag `-n` or `--no-color`. 

The first approach to solving the issue use the value from `color.NoColor` but as @pgimalac pointed out that would make the default value of `-n` or `--no-color` be dynamic. https://github.com/DataDog/datadog-agent/pull/26993/commits/2b06a9a1bfaa89de8b626597cc897163a08660fa

The second solution in this PR is just a patch to the problem. The real solution would be to change the configcheck endpoint to return a scrubbed JSON payload that the client latter display and render accordingly with or without color.

Reply:
In the end I decided to work on the more future proof solution https://github.com/DataDog/datadog-agent/pull/26993/commits/859c9273c1c72eb6404c38ac7b1f249b120459b1